### DISCUSSION TITLE
Trouble getting git-history to cooperate with scraped CSV

### DISCUSSION THREAD

**Discussion by scoates** [simonw/git-history#50] [Q&A] [answered]
Hi!

Thanks for setting this up.

I have a repository I've been scraping for quite a while and am just now getting around to actually tying to do something with it. I just made the [repository](https://github.com/scoates/scraped-data/tree/dev/emergency-rooms/quebec) public.

I managed to run `git-history` against [the .csv file](https://github.com/scoates/scraped-data/blob/dev/emergency-rooms/quebec/Releve_horaire_urgences_7jours.csv), but my SQLite .db file only has a `namespaces` table:

```
❯ pipenv run git-history file emergency_rooms.db scraped-data/emergency-rooms/quebec/Releve_horaire_urgences_7jours.csv --repo scraped-data --branch dev --csv
  [####################################]  11577/11577  100%          %
❯ sqlite3 emergency_rooms.db
SQLite version 3.32.3 2020-06-18 14:16:19
Enter ".help" for usage hints.
sqlite> .tables
namespaces
sqlite>
```

Is it obvious what I'm doing wrong here?

---

Side note: the docs say to use `--csv` but then they… don't in the example:
![](https://files.scoat.es/oSJWAQNy.png)

I can submit a patch for this later, but just wanted to be sure I wasn't missing some nuance here, too.

**Comment by scoates** [comment]
A tiny bit of extra info—my filesystem layout looks like this:

```
❯ find . | grep -v .git
.
./.envrc
./shell.nix
./Pipfile
./scraped-data
./scraped-data/emergency-rooms
./scraped-data/emergency-rooms/quebec
./scraped-data/emergency-rooms/quebec/capture
./scraped-data/emergency-rooms/quebec/README.md
./scraped-data/emergency-rooms/quebec/Releve_horaire_urgences_7jours.csv
./Pipfile.lock
./emergency_rooms.db
 ~/src/hospitals >
```

(where `scraped-data` is the repository linked above; that's why I specified it when I ran `git-history`)

**Reply by simonw** [reply]
Tried this just now and got the same result - definitely a bug. Might be related to accents or character encoding perhaps? I've not tested extensively there.

**Reply by scoates** [reply]
Glad it's not just my own clumsiness.

I also noticed that some of the columns that are otherwise numbers are marked `pas d'information disponible` (translates to `N/A`), in case that matters, too.

Maybe I could hack it with a `--convert`? (Edit: nope. `raise click.ClickException("Cannot use both --csv and --convert")`)

**Reply by scoates** [reply]
I dug in a little bit and found that `commit.tree.blobs` is always `[]` for me, [here](https://github.com/simonw/git-history/blob/1a2b0b82d73edbe4ffdbf00be1c984ee28cca589/git_history/cli.py#L30) (and it hits the `except` and `pass`es quietly). I don't know *why*, yet.

**Reply by scoates** [reply]
Patch(es) incoming.

**Comment by scoates** [comment, marked as answer]
https://github.com/simonw/git-history/pull/52
